sources/standard/inbox/__init__.py

```diff
@@ -0,0 +1,190 @@
+
+    Yields:
+        TDataItem: A dictionary containing the 'message_uid' of the collected email message.
+    """
+
+    last_message_num = initial_message_num.last_value
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select(folder, readonly=True)
+
+        criteria = [
+            f"(SINCE {start_date.strftime('%d-%b-%Y')})",
+            f"(UID {str(int(last_message_num))}:*)",
+            "(X-GM-RAW has:attachment)",
+        ]
+
+        if gmail_group:
+            logger.info(f"Load all emails for Group: {gmail_group}")
+            criteria.extend([f"(TO {gmail_group})"])
+
+        if filter_emails:
+            logger.info(f"Load emails only from: {filter_emails}")
+            if len(filter_emails) == 1:
+                filter_emails = filter_emails[0]
+            if isinstance(filter_emails, str):
+                criteria.append(f"(FROM {filter_emails})")
+            else:
+                email_filter = " ".join([f"FROM {email}" for email in filter_emails])
+                criteria.append(f"(OR {email_filter})")
+
+        uids = get_message_uids(client, criteria)
+        for i in range(0, len(uids), chucksize):
+            yield uids[i : i + chucksize]
+
+
+class ImapSource(FilesystemSource):
+    pass
+
+
+@dlt.transformer(
+    name="attachments",
+    write_disposition="merge",
+    merge_key="data_hash",
+    primary_key="data_hash",
+)
+def get_attachments_by_uid(
+    items: TDataItems,
+    storage_path: str,
+    host: str = dlt.secrets.value,
+    email_account: str = dlt.secrets.value,
+    password: str = dlt.secrets.value,
+    filter_by_mime_type: Sequence[str] = (),
+) -> TDataItem:
+    """Downloads attachments from email messages based on the provided message UIDs.
+
+    Args:
+        items (TDataItems): An iterable containing dictionaries with 'message_uid' representing the email message UIDs.
+        storage_path (str): The local folder path where attachments will be downloaded.
+        host (str, optional): The hostname of the IMAP server. Default is 'dlt.secrets.value'.
+        email_account (str, optional): The email account used to log in to the IMAP server. Default is 'dlt.secrets.value'.
+        password (str, optional): The password for the email account. Default is 'dlt.secrets.value'.
+        filter_by_mime_type (Sequence[str], optional): A sequence of MIME types used to filter attachments based on their content type. Default is an empty sequence.
+
+    Yields:
+        TDataItem: A dictionary containing the collected email information and attachment details.
+    """
+
+    with imaplib.IMAP4_SSL(host) as client:
+        client.login(email_account, password)
+        client.select()
+
+        for item in items:
+            message_uid = str(item["message_uid"])
+            msg = get_message(client, message_uid)
```

Comment:
I could not find any way, will give an other look at the imap protocol to see if I can find something.
But, as the body contains the whole email data, wouldn't it be better to get message by message and save the attachment storing just the metadata and so avoid using a lot of memory? Although email have a maximum of a few MBs so it would not hurt too much the memory.

Reply:
OK let's keep what we have